src/auth/views/AuthVerify.tsx

```diff
@@ -0,0 +1,67 @@
+import { AuthContainer } from "auth/views/components/AuthContainer";
+import { useAuthorize } from "auth/hooks/queries/useAuthorize";
+import React from "react";
+import { useHistory } from "react-router-dom";
+import { CircularProgress, Link, Stack, Typography } from "@mui/material";
+import VerifiedIcon from "@mui/icons-material/Verified";
+import CancelOutlinedIcon from "@mui/icons-material/CancelOutlined";
+
+export function AuthVerify() {
+  const history = useHistory();
+  const params = new URLSearchParams(history.location.search);
+
+  const { loading, error } = useAuthorize({
+    variables: {
+      code: params.get("code") ?? "",
+      id: params.get("id") ?? "",
+    },
+    onCompleted() {
+      history.push("/");
+    },
+  });
+
+  return (
+    <AuthContainer>
+      {loading && (
+        <Stack direction="column" alignItems="center" sx={{ mt: 7 }}>
+          <Typography variant="h4" sx={{ mb: 7 }}>
+            Logging in
+          </Typography>
+
+          <CircularProgress size={100} />
+        </Stack>
+      )}
+      {!loading && !error && (
+        <Stack direction="column" alignItems="center" sx={{ mt: 5 }}>
+          <Typography variant="h4" sx={{ mb: 5 }}>
+            Successfully logged in!
+          </Typography>
+          <VerifiedIcon sx={{ fontSize: "100px", mb: 5 }} color="success" />
+          <Link
+            variant="h5"
+            sx={{ cursor: "pointer", textAlign: "center" }}
+            onClick={() => history.replace("/")}
+          >
+            Not automatically redirected? Click this link to go to the
+            dashboard.
+          </Link>
```

Comment:
nit: I prefer this pattern rather than `onClick` (you probably don't need to cursor: pointer either because you get a proper link):

```tsx
import { Link as RouterLink } from "react-router-dom";

...

          <Link
            variant="h5"
            sx={{ cursor: "pointer" }}
            component={RouterLink}
            to={"/auth/link"}
            replace>

```

(also in a few other places, not repeated the comment)